mlflow/server/js/src/experiment-tracking/components/CompareRunView.js

```diff
@@ -37,6 +37,40 @@ export class CompareRunView extends Component {
     intl: PropTypes.shape({ formatMessage: PropTypes.func.isRequired }).isRequired,
   };
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      tableContainerWidth: null,
+      collapseParamBlock: false,
+      collapseMetricBlock: false,
+      onlyShowParamDiff: false,
+      onlyShowMetricDiff: false,
+    };
+    // onResizeHandler need access `this.state`, so bind `this`.
+    this.onResizeHandler = this.onResizeHandler.bind(this);
+    this.onTableBlockScrollHandler = this.onTableBlockScrollHandler.bind(this);
+
+    this.tableContainerRef = React.createRef();
+  }
+
+  onResizeHandler(e) {
+    const container = this.tableContainerRef.current;
+    if (container !== null) {
+      const containerWidth = container.clientWidth;
+      this.setState({ tableContainerWidth: containerWidth });
+    }
+  }
+
+  onTableBlockScrollHandler(e) {
+    const blocks = this.tableContainerRef.current.querySelectorAll('.compare-table .table-block');
+    for (let index = 0; index < blocks.length; ++index) {
```

Comment:
```suggestion
    for (let index = 0; index < blocks.length; ++index) {
```

Does `forEach((_, index) => {...})` work?